### ISSUE TITLE
Generate WADL for rest services

### ISSUE THREAD

**Issue by garcia-jj** [caelum/vraptor4#938]

**Comment by linyatis** [comment]
Great idea, @garcia-jj! If you need some help to test it in a real environment, just let me know :smiley:

**Comment by csokol** [comment]
maybe this support should be done in vraptor-restfulie plugin?

**Comment by garcia-jj** [comment]
@csokol Good point where to implement. vraptor-restfulie is for hipermedia, but vraptor core exposes rest services using `@Consumes` and with `Result.use(json())`.

I don't know what's the best place.

**Comment by linyatis** [comment]
Hummm, +1 @csokol.  
Guys, I think that vraptor-restfulie needs some docs enhancement, don't? I have not used, but looks very promising.

**Comment by linyatis** [comment]
I enjoy it in vraptor-restfulie because hypermedia is a pre-requisite of a truly rest service, don't?

**Comment by lucascs** [comment]
This should not be enabled by default, anyway, but we could add a WADL controller in the API which the user can enable, or have an annotation to specify which endpoints will go to the WADL response.

**Comment by flaviociaware** [comment]
PORTUGUESE:
Estamos desenvolvendo uma API Rest cujo o frontcontroller é em Vraptor por ser um excelente produto brasileiro. Porém, não conseguimos gerar interface para SWAGGER (documentador de apis rest).

Tentamos o GUJ (http://www.guj.com.br/t/vraptor-swagger-restapi/319575), um colega sugeriu usar o SOAPUI, mas ele importa WADL e novamente voltamos a estaca zero.

O Spring já faz isso e caso não consigamos encontrar uma forma de gerar a documentação da nossa Rest API teremos de migrar para Spring.

Nosso time de desenvolvedores pode auxiliar na criação dessa feature, mas precisamos de instrução de como proceder e uma previsão de quanto tempo implementariam esse recurso.

ENGLISH:
We are developing an API Rest whose FrontController is in VRaptor to be an excellent Brazilian product. However, we can not generate interface for SWAGGER (documenter of rest apis).

Try the GUJ (http://www.guj.com.br/t/vraptor-swagger-restapi/319575), a colleague suggested using the SOAPUI, but it matters WADL and again we go back to square one.

The Spring already does this and if we can not find a way to generate the documentation of our Rest API will have to migrate to Spring.

Our team of developers can help create this feature, but we need instruction on how to proceed and an estimate of how long it would implement this feature.

**Comment by flaviociaware** [comment]
PORTUGUESE:
Não sou arquiteto do VRaptor, mas seguindo a ideia do Swing, para documentar usando SWAGGER, anotações devem ser utilizadas para sinalizar o que existe nos controllers e no web.xml deveria estar o caminho de geração dos arquivos. Talvez isso se aplique também a geração do WADL.

ENGLISH:
I'm not VRaptor architect, but following the idea of Swing, to document using SWAGGER, annotations should be used to signal that exists in controllers and web.xml should be the path of the file generation. Perhaps this is also applies to generation WADL.
